compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/symbols/BallerinaObjectTypeSymbol.java

```diff
@@ -136,9 +136,17 @@ public BallerinaObjectTypeSymbol(CompilerContext context, ModuleID moduleID, BOb
 
     @Override
     public String signature() {
+        /*
+            When the object type is distinct we need generate the actual finite type as the signature. To avoid this for
+            anonymous types currently returns `var`
+         */
+        if (Symbols.isFlagOn(getBType().flags, Flags.ANONYMOUS)) {
+            return "var";
```

Comment:
@pubudu91 @nadeeshaan Is this okay.

Reply:
I have reverted LS changes and created an issue 
https://github.com/ballerina-platform/ballerina-lang/issues/29092